src/ReachSets/discretize.jl

```diff
@@ -1,310 +1,656 @@
-`Phi1Aabsdelta = P[1:n, (n+1):2*n]`, and `Phi2Aabs = P[1:n, (2*n+1):3*n]`.
-"""
-function discr_bloat_interpolation(cont_sys::InitialValueProblem{<:AbstractContinuousSystem},
-                                   δ::Float64,
-                                   approx_model::String,
-                                   pade_expm::Bool,
-                                   lazy_expm::Bool,
-                                   lazy_sih::Bool)
+The approximation model implemented in this function is such that there is no bloating,
+i.e. we don't bloat the initial states and don't multiply the input by the step
+size δ, as required for the dense time case.
 
-    sih = lazy_sih ? SymmetricIntervalHull : symmetric_interval_hull
+The transformations are:
 
-    A, X0 = cont_sys.s.A, cont_sys.x0
-    n = size(A, 1)
+- ``Φ ← \\exp^{Aδ}``,
+- ``Ω₀ ← ConvexHull(\\mathcal{X}_0, Φ\\mathcal{X}_0 ⊕ δU(0) ⊕ Eψ(U(0), δ) ⊕ E^+(\\mathcal{X}_0, δ))``,
+- ``V ← δU(k) ⊕ Eψ(U(k), δ)``.
 
-    # compute matrix ϕ = exp(Aδ)
-    if lazy_expm
-        ϕ = SparseMatrixExp(A*δ)
+Here we allow ``U`` to be a sequence of time varying non-deterministic input sets.
+
+For the definition of the sets ``Eψ`` and ``E^+`` see [1]. The  "backward" method
+uses ``E^-``.
+
+[1] Frehse, Goran, et al. "SpaceEx: Scalable verification of hybrid systems."
+International Conference on Computer Aided Verification. Springer, Berlin,
+Heidelberg, 2011.
+"""
+function discretize_interpolation(𝑆::InitialValueProblem{<:AbstractContinuousSystem},
+                                   δ::Float64;
+                                   algorithm::String="forward",
+                                   exp_method::String="base",
+                                   sih_method::String="concrete")
+
+    if sih_method == "concrete"
+        sih = symmetric_interval_hull
+    elseif sih_method == "lazy"
+        sih = SymmetricIntervalHull
     else
-        if pade_expm
-            ϕ = padm(A*δ)
-        else
-            ϕ = expmat(Matrix(A*δ))
-        end
+        throw(ArgumentError("the method $sih_method is unknown"))
     end
 
-    # early return for homogeneous systems
-    if cont_sys isa IVP{<:LinearContinuousSystem}
-         Ω0 = CH(X0, ϕ * X0)
-        return DiscreteSystem(ϕ, Ω0)
-    end
-    U = inputset(cont_sys)
-    inputs = next_set(U, 1)
+    # unrwap coefficient matrix and initial states
+    A, X0 = 𝑆.s.A, 𝑆.x0
+
+    # compute matrix ϕ = exp(Aδ)
+    ϕ = exp_Aδ(A, δ, exp_method=exp_method)
 
     # compute the transformation matrix to bloat the initial states
-    if lazy_expm
-        P = SparseMatrixExp([abs.(A*δ) sparse(δ*I, n, n) spzeros(n, n);
-                             spzeros(n, 2*n) sparse(δ*I, n, n);
-                             spzeros(n, 3*n)])
-        Phi2Aabs = sparse(get_columns(P, (2*n+1):3*n)[1:n, :])
+    Phi2Aabs = Φ₂(abs.(A), δ, exp_method=exp_method)
+
+    if algorithm == "forward"
+        Einit = sih(Phi2Aabs * sih((A * A) * X0))     # use Eplus
```

Comment:
```suggestion
        Einit = sih(Phi2Aabs * sih((A * A) * X0))     # use E⁺
```

Reply:
nice, how did you do that? doesnt work in Atom :(

Reply:
Now you can find out :wink:
```julia
help?> ⁺
"⁺" can be typed by \^+<tab>
```